lms/djangoapps/instructor/static/instructor/ReportPopup/components/Popup/PopupModal.jsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Modal, Button, Hyperlink} from '@edx/paragon/static';
+
+export default class PopupModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.handleDownload = this.handleDownload.bind(this);
+    }
+
+    handleDownload() {
+        window.open(`/media/${this.props.reportPath}`, '_blank');
+    }
+
+    render() {
+        let buttons = [];
+
+        if (this.props.reportPath != null) {
+            buttons.push((
+                <Button label={gettext("Download CSV")} onClick={this.handleDownload}/>
+            ));
+        }
+
+        let progress = null;
+        if (this.props.reportPath == null) {
+            if (this.props.inProgress) {
+                progress = (
+                    <div class="title">
+                        <b>{gettext("Your report is being created...")}</b>
+                    </div>
+                );
+            }
+        }
+
+        let msg = null;
+        if (this.props.error != null) {
+            msg = (
+                <div className="msg msg-error">{this.props.error}</div>
+            );
+        } else if (this.props.message != null) {
+            msg = (
+                <div
+                    className={`msg ${this.props.succeeded ? "msg-success" : "msg-warning"}`}>{this.props.message}</div>
```

Comment:
Use the ``classnames`` library [here](https://github.com/JedWatson/classnames#usage).